onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -788,7 +788,9 @@ ONNXTensorElementDataType GetTensorElementType(const ONNX_NAMESPACE::TensorProto
   return CApiElementTypeFromProtoType(tensor_proto.data_type());
 }
 
-ONNX_NAMESPACE::TensorProto TensorToTensorProto(const Tensor& tensor, const std::string& tensor_proto_name) {
+ONNX_NAMESPACE::TensorProto TensorToTensorProto(
+    const Tensor& tensor, const std::string& tensor_proto_name,
+    bool save_as_external_data, const PathString& external_data_path) {
```

Comment:
I don't think we need this change here. Like I mentioned before let's not add in the external data case just yet.

Reply:
BTW, I think we already have a way to save as external data in ort...

Reply:
Some explanation here: this is to support the legacy checkpoint api implementation (which use external data by default). 

As talked offline, let's avoid reusing those legacy logic in checkpointing.h/cc, instead write our own logic.